### DISCUSSION TITLE
Parabolic inflow 3D cases

### DISCUSSION THREAD

**Discussion by rubensamarojr** [Xiangyu-Hu/SPHinXsys#123] [Q&A] [answered]
Hi there,

The 3D tests [test_3d_T_shaped_pipe_stl](https://github.com/Xiangyu-Hu/SPHinXsys/tree/master/tests/3d_examples/test_3d_T_shaped_pipe_stl) and [test_3d_artery_with_stenosis_velocityBC](https://github.com/Xiangyu-Hu/SPHinXsys/tree/master/tests/3d_examples/test_3d_artery_with_stenosis_velocityBC) run correctly for constant inflow velocity. 
However, applying a paraboloid profile inflow, the case does not work correctly, with some leaking of particles near the outlet boundary.

Is this implementation not ready for this condition (spatial-variable inflow) yet or could I be doing something wrong ?

**Comment by Xiangyu-Hu** [comment]
Hi 
This case is contributed by @alundilong . You can ask him for details.

**Comment by rubensamarojr** [comment]
Hi,

Thanks @Xiangyu-Hu. I increased the length of the pipe near the inflow, so that a paraboloid flow profile should be created along, even for a uniform and cte inflow. However, a free-slip BC is observed and the flow is uniform along the pipe. Is there a way to set no-slip BC ?

@alundilong, please see below the change I made to the inflow of the case **test_3d_artery_with_stenosis_velocityBC**:

	class EmitterBufferInflowConditionWithFace : public VelocityInflowConditionWithFace
	{
		Real u_ave_, u_ref_, t_ref_;

	public:
		EmitterBufferInflowConditionWithFace(FluidBody &body, BodyRegionByCellsWithFace &body_part)
			: VelocityInflowConditionWithFace(body, body_part),
			  u_ave_(0), u_ref_(U_f), t_ref_(2.0) {}

		Vecd defineVelocityProfile(Vecd &position, Vecd &velocity) override
		{
			// Uniform
	///		Vecd vel = u_ave_ * inlet_direction.normalize();
			
			// Paraboloid
			Real diameter = 0.003;
			Real Yc = position[1] - 0.0005;
			Real Zc = position[2] + 0.0107;
			Real ux = (-4.0 * (Yc * Yc + Zc * Zc) / (diameter * diameter) + 1.0) * u_ave_;
			Vecd vel = ux * inlet_direction.normalize();
			return vel;
		}

		void setupDynamics(Real dt = 0.0) override
		{
			Real run_time = GlobalStaticVariables::physical_time_;
			u_ave_ = run_time < t_ref_ ? 0.5 * u_ref_ * (1.0 - cos(Pi * run_time / t_ref_)) : u_ref_;
		}
	};

As I explained, some particles leak near the outlet boundary. Do you have any idea if we can try to change the outlet function to avoid this ?

**Reply by Xiangyu-Hu** [reply, marked as answer]
We have take this case out the test set because the added relevant source does not consistent with main library. You can try to use the method in the main library for this problem.
